prism/src/main/java/org/apache/falcon/jdbc/MonitoringJdbcStateStore.java

```diff
@@ -187,14 +187,7 @@ public void putPendingInstances(String entity, String clusterName, Date nominalT
         EntityManager entityManager = getEntityManager();
         Query q = entityManager.createNamedQuery(PersistenceConstants.GET_ALL_PENDING_INSTANCES);
         List result = q.getResultList();
-
-        try {
-            if (CollectionUtils.isEmpty(result)) {
-                return null;
-            }
-        } finally{
-            entityManager.close();
-        }
+        entityManager.close();
```

Comment:
Nit: checked the usage of this method. You might have to modify checkPendingInstanceAvailability() of EntitySLAMonitoringService

Also getAllPendingInstances is called twice int that method. It is expensive as it is a DB call. While you are at it, please modify that too.